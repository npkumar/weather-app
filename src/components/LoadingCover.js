import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Typography, CircularProgress } from "@material-ui/core";

const useStyles = makeStyles({
  root: {
    height: "100vh",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    textAlign: "center",
    fontSize: 24,
    fontWeight: "bold",
    marginLeft: 16,
  },
});

const LoadingCover = () => {
  const classes = useStyles();
  return (
    <Typography component="div" className={classes.root}>
      <CircularProgress color="secondary" />
      <Typography
        data-testid="loadingText"
        component="div"
        className={classes.text}
      >
        Loading...
      </Typography>
    </Typography>
  );
};

export default LoadingCover;
