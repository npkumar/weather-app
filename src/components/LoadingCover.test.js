import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import LoadingCover from "./LoadingCover";

describe("LoadingCover", () => {
  afterEach(() => cleanup());

  it("renders", () => {
    const { asFragment } = render(<LoadingCover />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("has loading text", () => {
    const { getByTestId } = render(<LoadingCover />);
    expect(getByTestId("loadingText")).toHaveTextContent("Loading...");
  });
});
