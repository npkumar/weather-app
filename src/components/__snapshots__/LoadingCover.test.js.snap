// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingCover renders 1`] = `
<DocumentFragment>
  <div
    class="MuiTypography-root makeStyles-root-1 MuiTypography-body1"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
    <div
      class="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
      data-testid="loadingText"
    >
      Loading...
    </div>
  </div>
</DocumentFragment>
`;
